lib/galaxy/objectstore/irods.py

```diff
@@ -419,18 +419,26 @@ def _push_to_irods(self, rel_path, source_file=None, from_string=None):
                 log.debug("Wanted to push file '%s' to iRODS collection '%s' but its size is 0; skipping.", source_file, rel_path)
                 return True
 
+            # Create sub-collection first
```

Comment:
Creating a sub-folder and a data object must happen in both if/else clause below. Moved these steps before if/else clause.